src/app/components/login/login.component.ts

```diff
@@ -1,44 +1,37 @@
 import { Component, OnInit, ChangeDetectionStrategy } from '@angular/core';
-import {UserAuthService} from '../../services/user-auth.service';
-import { AngularFire, AuthProviders } from 'angularfire2';
 import {Observable} from 'rxjs/Observable';
 import 'rxjs';
-import {Store} from '@ngrx/store';
-import {AppState} from '../../reducers/';
-import * as userAuth from '../../actions/user-auth';
-import * as fromUserAuth from '../../reducers/user-auth';
-import * as fromRoot from '../../reducers';
-import { User } from '../../models/user';
+import { Router } from '@angular/router';
+import { Store } from '@ngrx/store';
+import { AppState, getLoginState } from '../../reducers/index';
```

Comment:
Yeah, we can do it the way you want to but I wanted to explicitly import the methods that I want.
rather than bring all the methods from the reducer taht are not needed in this component.